pkg/utils/deployer.go

```diff
@@ -78,14 +78,18 @@ func DeployableByHub(clusterLister clusterlisters.ManagedClusterLister, clusterI
 		return false, fmt.Errorf("failed to find a ManagedCluster declaration in namespace %s", dedicatedNamespace)
 	}
 
-	if mcls[0].Spec.SyncMode == clusterapi.Pull || !mcls[0].Status.AppPusher {
-		msg := "set syncMode as Pull"
-		if !mcls[0].Status.AppPusher {
-			msg = "disabled AppPusher"
-		}
-		klog.V(5).Infof("ManagedCluster %s with uid=%s has %s", klog.KObj(mcls[0]), mcls[0].UID, msg)
+	if mcls[0].Spec.SyncMode == clusterapi.Pull {
+		klog.V(5).Infof("ManagedCluster %s with uid=%s has %s", klog.KObj(mcls[0]), mcls[0].UID, "set syncMode as Pull")
 		return false, nil
 	}
+	if mcls[0].Status.AppPusher == nil {
+		return false, fmt.Errorf("unknown AppPusher")
+	} else {
```

Comment:
Remove `else`. We've already returned error in L86.

Reply:
else for handle AppPusher=false case, cannot remove

Reply:
remove keyword `else`, but keep its logic.

```go
if xxx {
    return xxx
}

xxxx

```

Reply:
done